heapprofiler: broken query dump mechanism in SQL process

**Describe the problem**

The periodic query dump mechanism was added in https://github.com/cockroachdb/cockroach/pull/66901. 
I'm seeing the following crash in SQL pods on CC whenever the process is under memory pressure:

```
E210930 18:12:34.462407 81 1@util/log/logcrash/crash_reporting.go:140  [-] 20757  a panic has occurred!
E210930 18:12:34.462407 81 1@util/log/logcrash/crash_reporting.go:140  [-] 20757 +runtime error: invalid memory address or nil pointer dereference
E210930 18:12:34.462407 81 1@util/log/logcrash/crash_reporting.go:140  [-] 20757 +(1) attached stack trace
E210930 18:12:34.462407 81 1@util/log/logcrash/crash_reporting.go:140  [-] 20757 +  -- stack trace:
E210930 18:12:34.462407 81 1@util/log/logcrash/crash_reporting.go:140  [-] 20757 +  | runtime.gopanic
E210930 18:12:34.462407 81 1@util/log/logcrash/crash_reporting.go:140  [-] 20757 +  | 	/usr/local/go/src/runtime/panic.go:965
E210930 18:12:34.462407 81 1@util/log/logcrash/crash_reporting.go:140  [-] 20757 +  | runtime.panicmem
E210930 18:12:34.462407 81 1@util/log/logcrash/crash_reporting.go:140  [-] 20757 +  | 	/usr/local/go/src/runtime/panic.go:212
E210930 18:12:34.462407 81 1@util/log/logcrash/crash_reporting.go:140  [-] 20757 +  | runtime.sigpanic
E210930 18:12:34.462407 81 1@util/log/logcrash/crash_reporting.go:140  [-] 20757 +  | 	/usr/local/go/src/runtime/signal_unix.go:734
E210930 18:12:34.462407 81 1@util/log/logcrash/crash_reporting.go:140  [-] 20757 +  | github.com/cockroachdb/cockroach/pkg/sql.(*SessionRegistry).SerializeAll
E210930 18:12:34.462407 81 1@util/log/logcrash/crash_reporting.go:140  [-] 20757 +  | 	/go/src/github.com/cockroachdb/cockroach/pkg/sql/exec_util.go:1824
E210930 18:12:34.462407 81 1@util/log/logcrash/crash_reporting.go:140  [-] 20757 +  | github.com/cockroachdb/cockroach/pkg/server/heapprofiler.(*ActiveQueryProfiler).takeQueryProfile
E210930 18:12:34.462407 81 1@util/log/logcrash/crash_reporting.go:140  [-] 20757 +  | 	/go/src/github.com/cockroachdb/cockroach/pkg/server/heapprofiler/activequeryprofiler.go:163
E210930 18:12:34.462407 81 1@util/log/logcrash/crash_reporting.go:140  [-] 20757 +  | github.com/cockroachdb/cockroach/pkg/server/heapprofiler.(*ActiveQueryProfiler).MaybeDumpQueries
E210930 18:12:34.462407 81 1@util/log/logcrash/crash_reporting.go:140  [-] 20757 +  | 	/go/src/github.com/cockroachdb/cockroach/pkg/server/heapprofiler/activequeryprofiler.go:102
E210930 18:12:34.462407 81 1@util/log/logcrash/crash_reporting.go:140  [-] 20757 +  | github.com/cockroachdb/cockroach/pkg/server.startSampleEnvironment.func1
E210930 18:12:34.462407 81 1@util/log/logcrash/crash_reporting.go:140  [-] 20757 +  | 	/go/src/github.com/cockroachdb/cockroach/pkg/server/server.go:2621
E210930 18:12:34.462407 81 1@util/log/logcrash/crash_reporting.go:140  [-] 20757 +  | github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx.func2
E210930 18:12:34.462407 81 1@util/log/logcrash/crash_reporting.go:140  [-] 20757 +  | 	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:442
E210930 18:12:34.462407 81 1@util/log/logcrash/crash_reporting.go:140  [-] 20757 +  | runtime.goexit
E210930 18:12:34.462407 81 1@util/log/logcrash/crash_reporting.go:140  [-] 20757 +  | 	/usr/local/go/src/runtime/asm_amd64.s:1371
E210930 18:12:34.462407 81 1@util/log/logcrash/crash_reporting.go:140  [-] 20757 +Wraps: (2) runtime error: invalid memory address or nil pointer dereference
E210930 18:12:34.462407 81 1@util/log/logcrash/crash_reporting.go:140  [-] 20757 +Error types: (1) *withstack.withStack (2) runtime.errorString
E210930 18:12:34.464429 81 util/log/logcrash/crash_reporting.go:323  [-] 20758  Queued as error 114fe4c99359483d98fad29c50010e79
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x363802b]

goroutine 81 [running]:
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).Recover(0xc001282000, 0x8c7ec20, 0xc0001b2010)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:231 +0x126
panic(0x4ae8f80, 0xb257630)
	/usr/local/go/src/runtime/panic.go:965 +0x1b9
github.com/cockroachdb/cockroach/pkg/sql.(*SessionRegistry).SerializeAll(0x0, 0x0, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/exec_util.go:1824 +0x6b
github.com/cockroachdb/cockroach/pkg/server/heapprofiler.(*ActiveQueryProfiler).takeQueryProfile(0xc001102cc0, 0x8c7ec20, 0xc0001b2010, 0x0, 0x1b8d7227, 0xed8e7f512, 0x0, 0x181e6e000, 0xc0001b2000)
	/go/src/github.com/cockroachdb/cockroach/pkg/server/heapprofiler/activequeryprofiler.go:163 +0x208
github.com/cockroachdb/cockroach/pkg/server/heapprofiler.(*ActiveQueryProfiler).MaybeDumpQueries(0xc001102cc0, 0x8c7ec20, 0xc0001b2010, 0x0, 0xc000d98000)
	/go/src/github.com/cockroachdb/cockroach/pkg/server/heapprofiler/activequeryprofiler.go:102 +0x105
github.com/cockroachdb/cockroach/pkg/server.startSampleEnvironment.func1(0x8c7ec20, 0xc0001b2010)
	/go/src/github.com/cockroachdb/cockroach/pkg/server/server.go:2621 +0x406
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx.func2(0xc001282000, 0x8c7ec20, 0xc0001b2010, 0x0, 0x0, 0xc001a72c80)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:442 +0xf3
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:434 +0x22b
```

**To Reproduce**

I reproduced this on CC's environment. Did not try to reproduce this locally.

Run queries with large sorts. An example would be `select trunc(random()*1000000000)::int as name from generate_series(1, 100000000) order by name;`. I also configured `sql.distsql.temp_storage.workmem` to a large number to push everything into memory where possible. 

**Expected behavior**

A segfault shouldn't occur.

I believe we need to pass in `tenantStatusServer.sessionRegistry` to `sampleEnvironmentCfg`:
https://github.com/cockroachdb/cockroach/blob/ac885449c1fe74de23ea5d82858b3aa3674496f9/pkg/server/tenant.go#L239-L250

cc: @abarganier @joshimhoff @cockroachdb/obs-inf-prs

--- a/pkg/server/tenant.go
+++ b/pkg/server/tenant.go
@@ -245,6 +245,7 @@ func StartTenant(
 		goroutineDumpDirName: args.GoroutineDumpDirName,
 		heapProfileDirName:   args.HeapProfileDirName,
 		runtime:              args.runtime,
+		sessionRegistry:      args.sessionRegistry,
 	}); err != nil {
 		return nil, "", "", err
 	}

